### PR TITLE
Add aqua tool installer cli

### PR DESCRIPTION
**Pull request by boris-smidt-klarrio**
## What does this PR do
It adds the Aqua cli installer. 
https://aquaproj.github.io/

## Standards checklist

- [x] The PR title is descriptive.
- [x] I have read `CONTRIBUTING.md`
- [x] *Optional:* I have tested the code myself
 
## For new steps

- [ ] *Optional:* Topgrade skips this step where needed
- [ ] *Optional:* The `--dry-run` option works with this step
- [x] *Optional:* The `--yes` option works with this step if it is supported by 
  the underlying command

If you developed a feature or a bug fix for someone else and you do not have the
means to test it, please tag this person here.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -300,6 +300,7 @@ fn run() -> Result<()> {
         runner.execute(Step::HomeManager, "home-manager", || unix::run_home_manager(&ctx))?;
         runner.execute(Step::Asdf, "asdf", || unix::run_asdf(&ctx))?;
         runner.execute(Step::Mise, "mise", || unix::run_mise(&ctx))?;
+        runner.execute(Step::Aqua, "aqua", || unix::run_aqua(&ctx))?;
```
Comment (SteveLauC) [inline src/main.rs:303]: Why limit this to UNIX, looks like Windows is also supported: https://aquaproj.github.io/docs/install

File: src/steps/os/unix.rs [hunk 2]
```diff
@@ -749,6 +749,16 @@ pub fn run_maza(ctx: &ExecutionContext) -> Result<()> {
     ctx.run_type().execute(maza).arg("update").status_checked()
 }
 
+pub fn run_aqua(ctx: &ExecutionContext) -> Result<()> {
```
Comment (SteveLauC) [inline src/steps/os/unix.rs:752]: Have you tried the dry run option, does it work with this new step?

```sh
$ topgrade --only aqua --dry-run
```

Also the `--yes` option, does Aqua have this option or something similar?

>  Optional: The --dry-run option works with this step
 Optional: The --yes option works with this step if it is supported by
the underlying command

**Review by SteveLauC — CHANGES_REQUESTED** [review]

File: src/steps/os/unix.rs (hunk 2 above)
Comment (boris-smidt-klarrio) [reply src/steps/os/unix.rs:752]: There isn't any -y or --yes on the update commands, So i don't think there is a dry run available.

File: src/main.rs (hunk 1 above)
Comment (boris-smidt-klarrio) [reply src/main.rs]: I've moved it to generic.rs

File: src/steps/os/unix.rs (hunk 2 above)
Comment (SteveLauC) [reply src/steps/os/unix.rs]: > There isn't any -y or --yes on the update commands

Get it, then we don't need to worry about this.

Dry run is another option, which is not related to --yes, when it is passed to Topgrade, Topgrade should simply echo what will be done and exit, does this PR implement it?

Comment (boris-smidt-klarrio) [reply src/steps/os/unix.rs]: There isn't a dry-run option either.

Comment (SteveLauC) [reply src/steps/os/unix.rs]: So 

```sh
$ topgrade --only aqua --dry-run
```

would actually do the update, right? We need to emulate this `--dry-run` option on the Topgrade's side, see:

https://github.com/topgrade-rs/topgrade/blob/a2375b48203322e39d36a6235db6aabdc4d3e5e0/src/self_update.rs#L20-L23

for an example.

**Review by SteveLauC — APPROVED** [review]
Thanks!

**Comment by boris-smidt-klarrio** [comment]
Thank you for the fast response it was a pleasure to contribute to topgrade!

**Pull request merged**